cranelift/control/src/chaos.rs

```diff
@@ -1,28 +1,66 @@
+use std::fmt::Display;
+
 use arbitrary::Arbitrary;
 
 /// The control plane of chaos mode.
 /// Please see the [crate-level documentation](crate).
 #[derive(Debug, Clone, Default)]
 pub struct ControlPlane {
     data: Vec<bool>,
+    fuel: Option<u32>,
 }
 
 impl Arbitrary<'_> for ControlPlane {
     fn arbitrary<'a>(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
         Ok(Self {
             data: u.arbitrary()?,
+            fuel: None,
         })
     }
 }
 
 impl ControlPlane {
+    /// Returns `true` if fuel wasn't activated in the first place, `false`
+    /// if there is no more fuel available.
+    fn consume_fuel(&mut self) -> bool {
+        match self.fuel {
+            None => true,               // fuel deactivated
+            Some(f) if f == 0 => false, // no more fuel
+            Some(ref mut f) => {
+                *f -= 1;
+                true
+            }
+        }
+    }
+
+    /// Set the maximum number of perturbations to be introduced with chaos
+    /// mode. Can be used to binary-search for a perturbation that triggered
+    /// a bug.
```

Comment:
I would describe this a little differently, because the intent is for this to be able to control other compiler mutations as well, not just those introduced by chaos-mode. "Set the fuel limit, which controls the number of mutations or optimizations that the compiler will perform before stopping. Currently applies to chaos perturbations but may apply to other optimization decisions in the future." or something like that?

Reply:
I've tried to incorporate the general information in there into the heading at the crate level documentation and linked to that from `ControlPlane::new` (which replaces `set_fuel`).